codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/StructureGenerator.java

```diff
@@ -113,7 +114,17 @@ public void renderStructure(Runnable runnable, boolean isInputStructure) {
                                 .orElse(memberSymbol);
                     }
 
-                    writer.write("$L $P", memberName, memberSymbol);
+                    String jsonTag = "";
+                    if (member.hasTrait(JsonNameTrait.ID)) {
+                        jsonTag = member.getMemberTrait(model, JsonNameTrait.class).get().getValue();
+                    }
+
+                    if (!jsonTag.isEmpty()) {
+                        writer.write("$L $P `json:\"$L\"`", memberName, memberSymbol, jsonTag);
+                    } else {
+                        writer.write("$L $P", memberName, memberSymbol);
+                    }
+
```

Comment:
Consider using optionals here instead of all this conditionals, something like:

```
writer.writeInline("$L $P", memberName, memberSymbol);

member.getMemberTrait(model, JsonNameTrait.class).map(JsonNameTrait::getValue)
      .map(jsonName -> writer.writeInline(" `json:\"$L\"`", jsonName));
                        
writer.write("");
```